### PR TITLE
Fix some types in `realization`

### PR DESCRIPTION
**Pull request by Goorzhel**

### CODE REVIEW

**Comment by yacoob** [comment]
Hey @Goorzhel; I'm done for the day today, but I'll do my best to review your outstanding PRs no later than Monday. Have a great weekend!

**Comment by yacoob** [comment]
Also: I've started a `dev` branch that contains some build-related fixes, and brings the versions of both tooling and dependencies forward. Can you please rebase your PRs on top of that? Thanks!

File: beancount-stubs/core/realization.pyi [hunk 1]
```diff
@@ -40,7 +40,7 @@ def realize(
 ) -> RealAccount: ...
 def postings_by_account(
     entries: Iterable[Directive],
-) -> list[dict[Account, TxnPosting | Directive]]: ...
+) -> dict[Account, list[Account, TxnPosting | Directive]]: ...
```
Comment (yacoob) [inline beancount-stubs/core/realization.pyi]: This should be just `TxnPosting | Directive`, no?

File: beancount-stubs/core/realization.pyi [hunk 2]
```diff
@@ -34,25 +34,25 @@ def get_or_create(
...
 def iterate_with_balance(
-    txn_postings: list[Posting | Directive],
+    txn_postings: Iterable[Posting | Directive],
 ) -> Iterator[tuple[Directive, list[Posting], Inventory, Inventory]]: ...
 def compute_balance(
     real_account: RealAccount, leaf_only: bool = ...
 ) -> Inventory: ...
 def find_last_active_posting(
-    txn_postings: list[Posting | Directive],
+    txn_postings: Iterable[Posting | Directive],
```
Comment (yacoob) [inline beancount-stubs/core/realization.pyi]: `find_last_active_posting` calls `reversed()` on `txn_postings`, so it needs to be a `Sequence`.

**Review by yacoob — CHANGES_REQUESTED** [review]
Looks good, two minor observations.

File: beancount-stubs/core/realization.pyi (hunk 1 above)
Comment (Goorzhel) [reply beancount-stubs/core/realization.pyi]: D'oh.

**Review by yacoob — APPROVED** [review]

**Pull request merged**